### PR TITLE
Fix create-layer function for compleseus

### LINKED ISSUE #15913: configuration-layer/create-layer doesn’t work with Compleseus/Vertico
#### Description :octocat:

configuration-layer/create-layer doesn’t work with Compleseus/Vertico

#### Reproduction guide :beetle:
- Start Emacs
- M-x: configuration-layer/create-layer 

*Observed behaviour:* :eyes: :broken_heart:
Symbol’s function definition is void: helm

*Expected behaviour:* :heart: :smile:
A new layer should be created

#### System Info :computer:
- OS: darwin
- Emacs: 28.0.91
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 1638bbf39)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(csv python
     (auto-completion :variables auto-completion-return-key-behavior 'complete auto-completion-tab-key-behavior 'complete auto-completion-enable-snippets-in-popup 'complete)
     bibtex emacs-lisp
     (compleseus :variables compleseus-engine 'vertico)
     (latex :variables latex-enable-folding t latex-enable-magic t latex-build-command "LatexMk")
     multiple-cursors
     (org :variables org-enable-roam-support t)
     org-reledmac pdf pandoc
     (ranger :variables ranger-show-preview t ranger-show-hidden t ranger-preview-delay 0.06 ranger-show-literal nil ranger-cleanup-eagerly t ranger-modify-header nil)
     (shell :variables shell-default-height 30 shell-default-position 'bottom)
     spell-checking personal-keybindings personal-org-settings
     (typography :variables typography-enable-typographic-editing t)
     treemacs)
```
- System configuration features: ACL GLIB GMP GNUTLS JPEG JSON LCMS2 LIBXML2 MODULES NATIVE_COMP NOTIFY KQUEUE NS PDUMPER PNG RSVG THREADS TIFF TOOLKIT_SCROLL_BARS XIM ZLIB

#### Backtrace :paw_prints:
```
Debugger entered--Lisp error: (void-function helm)
  (helm :sources helm-lp-source :prompt "Configuration layer path: ")
  (if (configuration-layer/layer-used-p 'ivy) (ivy-read "Configuration layer path: " (append current-layer-paths (list other-choice))) (helm :sources helm-lp-source :prompt "Configuration layer path: "))
  (let* ((current-layer-paths (mapcar #'(lambda (dir) (expand-file-name dir)) (let* ((var configuration-layer-private-layer-directory)) (if (memql var dotspacemacs-configuration-layer-path) (with-no-warnings dotspacemacs-configuration-layer-path) (setq dotspacemacs-configuration-layer-path (cons var dotspacemacs-configuration-layer-path)))))) (other-choice "Another directory...") (helm-lp-source (cons '(name . "Configuration Layer Paths") (cons (cons 'candidates (append current-layer-paths (list other-choice))) '((action lambda ... c))))) (layer-path-sel (if (configuration-layer/layer-used-p 'ivy) (ivy-read "Configuration layer path: " (append current-layer-paths (l
```

### PR DESCRIPTION
**Pull request by fnussbaum**
Fixes #15913

### CODE REVIEW

**Comment by bcc32** [comment]
FWIW, `ivy-read` with these arguments is equivalent to just `completing-read`, so I think the only special case needed is helm, here.

**Comment by fnussbaum** [comment]
Good point, thanks!

File: core/core-configuration-layer.el [hunk 1]
```diff
@@ -715,12 +715,12 @@ layer directory."
...
-                                       (append current-layer-paths
-                                               (list other-choice)))
-                           (helm :sources helm-lp-source
-                                 :prompt "Configuration layer path: ")))
+         (layer-path-sel (if (configuration-layer/layer-used-p 'helm)
+                             (helm :sources helm-lp-source
+                                   :prompt "Configuration layer path: ")
+                           (completing-read "Configuration layer path: "
+                                            (append current-layer-paths
+                                                    (list other-choice)))))
```
Comment (smile13241324) [inline core/core-configuration-layer.el:723]: Are you sure we do not need ivy read here too?

**Review by smile13241324 — CHANGES_REQUESTED** [review]
Please have a look at my review comment

**Review by smile13241324 — APPROVED** [review]

**Comment by smile13241324** [comment]
Ups missed the last comment

**Pull request merged**